vendor/bat-native-ads/src/bat/ads/internal/confirmations/confirmations_state.cc

```diff
@@ -643,6 +645,10 @@ bool ConfirmationsState::ParseAdRewardsFromDictionary(
     base::DictionaryValue* dictionary) {
   DCHECK(dictionary);
 
+  if (!ad_rewards_) {
```

Comment:
required for tests that use confirmations state that is not part of an integration test. https://github.com/brave/brave-browser/issues/12563 will remove ad rewards from confirmations